### PR TITLE
Produce block only once for a slot

### LINKED ISSUE #8625: [SSV] Fix cache behaviour when requesting blocks in parallel
There are scenarios where multiple operators may be connected to the same Beacon node,each of which requests the Beacon block proposal, in case they become the leader and must propose it on SSV. We have some cases in code, which cache data by slot for block proposal (with the usual assumption only one block proposal is requested per slot) so in case of two responses (one builder response and one fallback), there could be issues. On first look, the only two cases are in `ExecutionLayerBlockProductionManagerImpl` and in `FailoverValidatorApiHandler`.
 
More background on Discord and an issue occuring: https://discord.com/channels/697535391594446898/697539289042649190/1286291762167545958

### PR DESCRIPTION
**Pull request by StefanBratanov**
## PR Description
Essentially, return the same block if the createBlock endpoint is called twice. This means we no longer require the changes in #8766 

## Fixed Issue(s)
fixes #8625

## Documentation

- [X] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [X] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionLayerBlockProductionManagerImpl.java [hunk 1]
```diff
@@ -72,35 +67,42 @@ public ExecutionPayloadResult initiateBlockProduction(
...
+        blockSlotState.getSlot(),
+        __ -> {
+          if (attemptBuilderFlow) {
+            return executeBuilderFlow(
+                context, blockSlotState, requestedBuilderBoostFactor, blockProductionPerformance);
+          } else {
+            return executeLocalFlow(context, blockSlotState, blockProductionPerformance);
+          }
+        });
+  }
```
Comment (tbenr) [inline ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionLayerBlockProductionManagerImpl.java]: this is not guaranteed to be computed once. I think we need to synchronize.

Comment (StefanBratanov) [reply ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/ExecutionLayerBlockProductionManagerImpl.java]: Modified PR, so no longer applicable

File: beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java [hunk 2]
```diff
@@ -122,6 +122,9 @@ public class ValidatorApiHandler implements ValidatorApiChannel {
    */
   private static final int DUTY_EPOCH_TOLERANCE = 1;
 
+  private final Map<UInt64, SafeFuture<Optional<BlockContainerAndMetaData>>>
+      blockProductionFuturesBySlotCache = LimitedMap.createSynchronizedLRU(2);
+
   private final Map<UInt64, Bytes32> createdBlockRootsBySlotCache =
       LimitedMap.createSynchronizedLRU(2);
```
Comment (tbenr) [inline beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java]: I think we can get the rid of `createdBlockRootsBySlotCache` and use only the new `blockProductionFuturesBySlotCache`.

Comment (StefanBratanov) [reply beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java]: Yup, makes sense. Done.

File: beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java [hunk 3]
```diff
@@ -866,11 +883,13 @@ private List<ProposerDuty> getProposalSlotsForEpoch(final BeaconState state, fin
...
-    final Bytes32 locallyCreatedBlockRoot =
-        createdBlockRootsBySlotCache.get(blockContainer.getSlot());
-    return Objects.equals(blockRoot, locallyCreatedBlockRoot);
+  private boolean isLocallyCreatedBlock(final UInt64 slot) {
+    final SafeFuture<Optional<BlockContainerAndMetaData>> localBlockProduction =
+        localBlockProductionBySlotCache.get(slot);
+    if (localBlockProduction == null) {
+      return false;
+    }
+    return localBlockProduction.isCompletedNormally();
```
Comment (tbenr) [inline beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/ValidatorApiHandler.java]: I'd keep the blockRoot check here.

like 

```java
if (localBlockProduction == null) {
  return false;
}

if(! localBlockProduction.isCompletedNormally()) {
return false;
}
return localBlockProduction.join() .... .equals(.....));
```

**Review by tbenr — APPROVED** [review]
LGTM

**Pull request merged**